src/main/java/in/twizmwaz/cardinal/module/region/RegionBounds.java

```diff
@@ -0,0 +1,99 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package in.twizmwaz.cardinal.module.region;
+
+import in.twizmwaz.cardinal.module.region.exception.operation.RegionOperationException;
+import in.twizmwaz.cardinal.module.region.type.BlockRegion;
+import in.twizmwaz.cardinal.util.Numbers;
+import in.twizmwaz.cardinal.util.Vectors;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.bukkit.block.Block;
+import org.bukkit.util.BlockVector;
+import org.bukkit.util.Vector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+public class RegionBounds {
+
+  Vector min, max;
+
+  public static RegionBounds empty() {
+    return new RegionBounds(Vectors.max(), Vectors.min());
+  }
+
+  public static RegionBounds unbounded() {
+    return new RegionBounds(Vectors.min(), Vectors.max());
+  }
+
+  public boolean isBounded() {
+    return !(Double.isInfinite(this.min.getX())
+        || Double.isInfinite(this.max.getX())
+        || Double.isInfinite(this.min.getY())
+        || Double.isInfinite(this.max.getY())
+        || Double.isInfinite(this.min.getZ())
+        || Double.isInfinite(this.max.getZ()));
+  }
+
+  public Vector getRandom() {
+    return new Vector(Numbers.getRandom(min.getX(), max.getX()), Numbers.getRandom(min.getY(), max.getY()),
+        Numbers.getRandom(min.getZ(), max.getZ()));
+  }
+
+  public Vector getCenter() {
+    return min.midpoint(max);
+  }
+
+  public BlockRegion getCenterBlock() {
+    return new BlockRegion(getCenter());
+  }
+
+  public BlockVector blockAlign(Vector vector) {
+    return new BlockVector((int) vector.getX() + 0.5d, (int) vector.getY() + 0.5d, (int) vector.getZ() + 0.5d);
```

Comment:
That's not my point, you don't need to cast vector.getX() to an integer. Just use vector.getBlockX().